validation/genesis/genesis-predeploy_test.go

```diff
@@ -0,0 +1,195 @@
+package genesis
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"reflect"
+	"runtime"
+	"strconv"
+	"strings"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	. "github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum/core"
+	"github.com/stretchr/testify/require"
+)
+
+// TODO deduplicate this
+// perChainTestName ensures test can easily be filtered by chain name or chain id using the -run=regex testflag.
+func perChainTestName(chain *superchain.ChainConfig) string {
+	return chain.Name + fmt.Sprintf(" (%d)", chain.ChainID)
+}
+
+func TestGenesisPredeploys(t *testing.T) {
+	for _, chain := range OPChains {
+		if chain.SuperchainLevel == Standard || chain.StandardChainCandidate {
+			t.Run(perChainTestName(chain), func(t *testing.T) {
+				// Do not run in parallel
+				testGenesisPredeploys(t, chain)
+			})
+		}
+	}
+}
+
+// Invoke this with go test -timeout 0 ./validation/genesis -run=TestGenesisPredeploys -v
+// REQUIREMENTS:
+// pnpm and yarn, so we can prepare https://codeload.github.com/Saw-mon-and-Natalie/clones-with-immutable-args/tar.gz/105efee1b9127ed7f6fedf139e1fc796ce8791f2
+func testGenesisPredeploys(t *testing.T, chain *ChainConfig) {
+	chainId := chain.ChainID
+
+	vis, ok := ValidationInputs[chainId]
+
+	if !ok {
+		t.Skip("WARNING: cannot yet validate this chain (no validation metadata)")
+
+	}
+
+	monorepoCommit := vis.GenesisCreationCommit
+
+	// Setup some directory references
+	thisDir := getDirOfThisFile()
+	chainIdString := strconv.Itoa(int(chainId))
+	validationInputsDir := path.Join(thisDir, "validation-inputs", chainIdString)
+	monorepoDir := path.Join(thisDir, "../../../optimism-temporary")
+	contractsDir := path.Join(monorepoDir, "packages/contracts-bedrock")
+
+	// reset to appropriate commit, this is preferred to git checkout because it will
+	// blow away any leftover files from the previous run
+	executeCommandInDir(t, monorepoDir, exec.Command("git", "reset", "--hard", monorepoCommit))
+
+	// TODO unskip these, I am skipping to save time in development since we
+	// are not validating multiple chains yet
+	if false {
+		executeCommandInDir(t, monorepoDir, exec.Command("rm", "-rf", "node_modules"))
+		executeCommandInDir(t, contractsDir, exec.Command("rm", "-rf", "node_modules"))
+	}
+
+	// install dependencies
+	// TODO we expect this step to vary as we scan through the monorepo history
+	// so we will need some branching logic here
+	// executeCommandInDir(t, contractsDir, exec.Command("pnpm", "install", "--no-frozen-lockfile"))
+	executeCommandInDir(t, contractsDir, exec.Command("yarn", "install", "--no-frozen-lockfile"))
```

Comment:
We may need to use `nvm` depending on the version of node we need to build the monorepo.